etna/datasets/tsdataset.py

```diff
@@ -1034,21 +1036,23 @@ def get_level_dataset(self, target_level: str) -> "TSDataset":
         if target_level_index > current_level_index:
             raise ValueError("Target level should be higher in the hierarchy than the current level of dataframe!")
 
+        target_names = tuple(get_target_column_names(columns=self.columns))
+
         if target_level_index < current_level_index:
             summing_matrix = self.hierarchical_structure.get_summing_matrix(
                 target_level=target_level, source_level=self.current_df_level
             )
 
-            source_level_data = self[:, current_level_segments, "target"].values
-            target_level_data = source_level_data @ summing_matrix.T
-
-            target_level_segments = pd.MultiIndex.from_product(
-                [target_level_segments, ["target"]], names=["segment", "feature"]
+            target_level_df = get_level_dataframe(
+                df=self[:, current_level_segments, target_names],
```

Comment:
We actually can pass TSDataset here instead of dataframe, the line with "target_names" can also be calculated inside method(remove this parameter)

Reply:
It is better to pass dataframe, because inside `etna.datasets.utils` we can not import `TSDataset` as it will introduce circular import.